developer/src/kmc-kmn/test/test-compiler.ts

```diff
@@ -87,4 +74,35 @@ describe('Compiler class', function() {
       }
     }
   });
+
+  it('should compile a keyboard with visual keyboard', async function() {
+    const compiler = new KmnCompiler();
+    const callbacks = new TestCompilerCallbacks();
+    assert(await compiler.init(callbacks));
+    assert(compiler.verifyInitialized());
+
+    const fixtureDir = keyboardsDir + 'caps_lock_layer_3620/'
+    const infile = fixtureDir + 'source/caps_lock_layer_3620.kmn';
+    const kmxFixture = fixtureDir + '/binary/caps_lock_layer_3620.kmx';
+    const kvkFixture = fixtureDir + '/binary/caps_lock_layer_3620.kvk';
+
+    const kmxfile = __dirname + '/caps_lock_layer_3620.kmx';
+    const kvkfile = __dirname + '/caps_lock_layer_3620.kvk';
```

Comment:
This might be easier to understand if these would be called `resultingKmxFile` and `resultingKvkFile`.